packages/mapper/src/mapper.ts

```diff
@@ -1808,19 +1810,75 @@ export default class Mapper {
 
   collection_explain(collection: Collection, verbosity = 'queryPlanner'): Explainable {
     this._validateExplainableVerbosity(verbosity);
+
+    this.messageBus.emit(
+      'mongosh:api-call',
+      {
+        method: 'explain',
+        class: 'Collection',
+        db: collection._database._name,
+        coll: collection._name,
+        arguments: { verbosity }
+      }
+    );
+
     return new Explainable(this, collection, verbosity);
   }
 
+  private _emitExplainableApiCall(explainable: Explainable, methodName: string, methodArguments: any = {}): void {
+    this.messageBus.emit(
+      'mongosh:api-call',
+      {
+        method: methodName,
+        class: 'Explainable',
+        db: explainable._collection._database._name,
+        coll: explainable._collection._name,
+        arguments: methodArguments
+      }
+    );
+  }
+
   explainable_getCollection(explainable: Explainable): Collection {
+    this._emitExplainableApiCall(explainable, 'getCollection');
```

Comment:
@lrlna i'd like to do the same for collection and db methods, is much more readable like this. What do you think? do you see any issue with this?

Reply:
as long as you don't just pass on the `...arguments` from the emitting function, I think it will be ok.

Reply:
That should be fine, should be easier also to review.

Reply:
I hope